### ISSUE TITLE
Missing Uniswap trades

### ISSUE THREAD

**Issue by a7e8** [rotki/rotki#2614]
## Problem Definition

I see the first Uniswap trade for an account but not the 2nd, 3rd or 4th.

## Logs

n/a

### System Description

Operating system: macOS
Rotki version: 1.15.1

**Comment by LefterisJP** [comment]
Is the trade in v1? v1 is not supported at the moment.

 Can you DM me the account you expect to see the trade in? There is no way we can reproduce this without any more info.

**Comment by Blackswan122** [comment]
Yes I would love Rotki to drag my UNI V1 trades in.

**Comment by a7e8** [comment]
No it's v2. I'll send you the account.

**Comment by yabirgb** [comment]
What I found is that the subgraph for uniswap V2 has, in some swaps (`swap`),  the same address in the `to` and `sender` fields. I'm not sure but I believe has to do with how the router works and how swaps are stored. 

The thing is that the subgraph is not in sync with the repo chages where a [new field](https://github.com/Uniswap/uniswap-v2-subgraph/blob/master/src/mappings/core.ts#L491) has been added, `from`, and is the EOA creating the transaction. As per my tests this field `form` contains the address we should be querying in those cases. To verify this I queried [this](https://thegraph.com/explorer/subgraph/benesjan/uniswap-v2) graph that seems to be a copy of the one in the uniswaps's repo and, as I expected, the from address is in there and contains the info we needed.

So what I believe will be needed is to ask why the official graph is not updated, use this subgraph if no info is provided or if we don't trust it, see how to deploy a fork of the official one.

Also I queried for the issue by @LefterisJP https://github.com/Uniswap/uniswap-v2-subgraph/issues/85 and seems to be fixed
